server/app/controllers/api/follows_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Api::FollowsController < Api::ApplicationController
+  before_action :set_teacher
+  before_action :authenticate, only: %i(create destroy)
+
+  def create
+    follow = current_user.follows.new(teacher: @teacher)
+
+    if follow.save
+      json_response(message: "You are now following this teacher.",
+                    status: :created)
+    else
+      error_response(message: "Failed to follow the teacher.",
+                     status: :unprocessable_entity)
+    end
+  end
+
+  def destroy
+    follow = current_user.follows.find_by(teacher: @teacher)
+
+    if follow&.destroy
+      json_response(message: "You have unfollowed this teacher.", status: :ok)
+    else
+      error_response(message: "Failed to unfollow the teacher.",
+                     status: :not_found)
+    end
+  end
+
+  private
+
+  def set_teacher
+    @teacher = Teacher.find(params[:teacher_id])
+  rescue ActiveRecord::RecordNotFound
+    error_response(message: "Teacher not found", status: :not_found)
+  end
```

Comment:
```suggestion
  def set_teacher
    @teacher = Teacher.find_by teacher_id: params[:id]
  end
```
có cần đoạn này k á
  rescue ActiveRecord::RecordNotFound
    error_response(message: "Teacher not found", status: :not_found)

Reply:
Có á để bắt lỗi ko tìm đc teacher. Bth cũng có bắt nè
![image](https://github.com/user-attachments/assets/51fb1a02-1149-4922-8001-6cd51011925c)